File: dnmcForms/dnmcForms/Models/Field.swift
import Foundation

// MARK: - Field Model
struct Field: Identifiable, Codable {
    var id = UUID()
    let uuid: String
    let name: String
    let label: String
    let type: FieldType
    let options: [String]?
    let required: Bool?

    init(uuid: String, name: String, label: String, type: String, options: [String]?, required: Bool?) {
        self.uuid = uuid
        self.name = name
        self.label = label
        self.options = options
        self.required = required
        self.type = FieldType(rawValue: type) ?? .text
    }
}

// MARK: - Field Type Enum
enum FieldType: String, Codable {
    case description
    case dropdown
    case text
    case number
}

File: app/src/main/java/com/example/dnmcformsandroid/models/Field.kt
package com.example.dnmcformsandroid.models

import com.google.gson.annotations.SerializedName

enum class FieldType(val type: String) {
    @SerializedName("text") TEXT("text"),
    @SerializedName("number") NUMBER("number"),
    @SerializedName("dropdown") DROPDOWN("dropdown"),
    @SerializedName("description") DESCRIPTION("description")
}

data class Field(
    val uuid: String,
    val name: String,
    val label: String,
    val type: String,
    val options: List<String>? = null,
    val required: Boolean = false
)